cosmos/converter.py

```diff
@@ -370,3 +378,93 @@ def _add_dbt_project_hash_to_dag_docs(self, dag: DAG | None) -> None:
             logger.debug(f"Appended dbt project hash {dbt_project_hash} to DAG {dag.dag_id} documentation")
         except Exception as e:
             logger.warning(f"Failed to append dbt project hash to DAG documentation: {e}")
+
+    def _store_cosmos_telemetry_metadata_on_dag(  # noqa: C901
+        self,
+        dag: DAG | None,
+        render_config: RenderConfig,
+        execution_config: ExecutionConfig,
+        project_config: ProjectConfig,
+        operator_args: dict[str, Any],
+        initial_load_method: LoadMode,
+    ) -> None:
+        """
+        Store Cosmos configuration metadata on the DAG for telemetry purposes.
+
+        This metadata is used by the DAG run listener to emit telemetry metrics
+        about how Cosmos is configured and used.
+
+        :param dag: The Airflow DAG to store metadata on. If None, no action is taken.
+        :param render_config: The render configuration
+        :param execution_config: The execution configuration
+        :param project_config: The project configuration
+        :param operator_args: The operator arguments
+        :param initial_load_method: The load method specified by the user (before automatic resolution)
+        """
+        if dag is None:
+            return
+
+        metadata = {}
+
+        # Compute each metric individually with error handling
+        try:
+            metadata["used_automatic_load_mode"] = initial_load_method == LoadMode.AUTOMATIC
+        except Exception as e:
+            logger.warning(f"Failed to compute used_automatic_load_mode: {e}")
+
+        try:
+            metadata["actual_load_mode"] = str(self.dbt_graph.load_method.value)
+        except Exception as e:
+            logger.warning(f"Failed to compute actual_load_mode: {e}")
+
+        try:
+            invocation_mode = None
+            if execution_config.invocation_mode:
+                invocation_mode = str(execution_config.invocation_mode.value)
+            elif render_config.invocation_mode:
+                invocation_mode = str(render_config.invocation_mode.value)
+            metadata["invocation_mode"] = invocation_mode
+        except Exception as e:
+            logger.warning(f"Failed to compute invocation_mode: {e}")
+
+        try:
+            install_deps = operator_args.get("install_deps")
```

Comment:
Cosmos DAG parsing should not rely on `install_deps` from `operator_args`. It should rely exclusively in the `render_config` or `project_config` one